### DISCUSSION TITLE
Feedback for rotki improvements

### DISCUSSION THREAD

**Discussion by roadbridgeaward** [rotki/rotki#5635] [Archive]
Reason for not to buy rotki premium

rotki is a wonderful initiative, and most importantly it addresses a super important and steering need: “to have an accurate 360 presentation of all of the users assets everywhere in one place”. This is a user need that if addressed fully, it can mean the difference between a well informed user taking well informed profitable investment decisions and a user taking bad investment decisions. 

While I understand well that the user needs addressed by rotki are challenging by nature (away from rotki itself), I think it will be very useful if I expressed the reasons that I think repels a given user from paying for the premium subscription. This will be useful for both the developers of rotki to know and for the community to open this subject in a positive way aiming at further making the rotki project more useful and hence more widespread which will in turn cause massive number of subscriptions into premium which will cause more investment in the development effort and this will go into a loop that makes this project on the top of services in its field.
Here are the reasons I think:
1- Some important financial aspects on some exchanges are not considered which affects the reports and the statistics and in some other areas in rotki which in turn causes the generated report not to be a real 360 view of user assets on a given exchange. Ideally, if the user exported / extracted a 100% full statement file from his exchange across all products and every possible operation of any type of any time, this needs to be fully and accurately exist in rotki.

2- The fact that the statistics does not include history before running rotki makes investment decisions involving inputs needed from the past impossible.

3- The lack of a full ledger (that is exposed in the UI) in one and only one place representing 100% of any debit or credit operation that happened anywhere anytime.

Please feel free to further elaborate on this list and to criticize any of the reasons I mentioned so we have a solid outcome. Thank you all in advance and wishing rotki and rotki team the best.

**Comment by LefterisJP** [comment]
Thank you for your feedback.

We have issues trying to address all of those concerns you have:

1. Every exchange is different and adds stuff all the time. New exchanges pop up, old ones die. Supporting every single thing every exchange creates is a herculean task. If you want something specific for an exchange make an issue and we will handle it when we can.
2. https://github.com/rotki/rotki/issues/1379
3. You mean a unified history view? We are considering going towards that actually: https://github.com/rotki/rotki/issues/2469

Now let me give you two reasons to buy premium:
1. It's currently really cheap. This will change.
2. You want us to keep developing rotki and eventually address those issues. Software development costs money.

If you have any other actionable feedback (like please implement feature XXX for exchange YYY) feel free to open an issue.

**Reply by roadbridgeaward** [reply]
> Thank you for your feedback.
> 
> We have issues trying to address all of those concerns you have:
> 
> 1. Every exchange is different and adds stuff all the time. New exchanges pop up, old ones die. Supporting every single thing every exchange creates is a herculean task. If you want something specific for an exchange make an issue and we will handle it when we can.

Yes, I totally agree with you.

The issue is that we are after a usable software at the end of the day. If rotki will give only a partial presentation of the assets then (due to lacking of some number of some transactions performed on the exchange) then the user cannot depend fully on rotki on making investment decisions - something that will reduce the number of users who will seriously depend on rotki and hence reducing the number of premium subscribers.

Yes, this is a herculean task as you mentioned but my point is we have two approaches of thinking here:
a- rotki is a 100% accurate representation of use assets in one place: then it is more important to focus on less number of exchanges and fully support them guaranteeing an accurate representation of user assets positioning rotki as a reliable system based on which the user can make critical investment decisions., or
b- rotki just gives an approximate (not guaranteed to be 100% accurate) representation of user assets. In  this case user can have a general idea about his assets using rotki but cannot fully depend on rotki for making well informed investment decision. Supporting as many diverse exchanges as possible (which each exchange not fully supported) is a viable option in this case I think.

I do not know, but I wonder what is the best direction to invest development effort on: option a or option b?

In the realm of option a, partial support of an exchange is a waste of effort I think because we are after a black and white scenario here: you either have an accurate representation to rely on or not.

> 2. [Use historical data from before user started using the app in statistics (and elsewhere) #1379](https://github.com/rotki/rotki/issues/1379)
> 3. You mean a unified history view? We are considering going towards that actually: [Consider redesign of History section #2469](https://github.com/rotki/rotki/issues/2469)
> 
> Now let me give you two reasons to buy premium:
> 
> 1. It's currently really cheap. This will change.
> 2. You want us to keep developing rotki and eventually address those issues. Software development costs money.

Yes, I totally agree with you.

> If you have any other actionable feedback (like please implement feature XXX for exchange YYY) feel free to open an issue.

**Comment by LefterisJP** [comment]
> The issue is that we are after a usable software at the end of the day

The definition of usable is different for each user and their usecase, their portfolio and the platforms, chains, exchanges, features of exchanges, protocols etc. that they use.

We strive to make rotki as useful as possible for the majority of our users and constantly improve it. We ourselves are heavy users of the software.

We already  have thousands of happy free and paying users. We want more of you guys to be happy and prioritize accordingly.

Again if you have something **specific** that is missing create an issue about it. If there is one, add a +1 comment on it and we will work on then when we can depending on resources and priorities.

**Reply by LefterisJP** [reply]
Oh and by the way if something is not automatically tracked we allow users to add it as a manual balance until we manage to do so automatically: https://rotki.readthedocs.io/en/latest/usage_guide.html?highlight=Manual#adding-manual-balances

For more questions about rotki feel free to join our discord https://discord.gg/aGCxHG7

(Would need to have DMs open as there is a captcha challenge)

**Reply by roadbridgeaward** [reply]
Thank you so much for this effort
